src/test/scala/alu/ALUSpecFormal.scala

```diff
@@ -0,0 +1,24 @@
+package alu
+
+import chisel3._
+import chiseltest._
+import chiseltest.formal._
+import org.scalatest.freespec.AnyFreeSpec
+import chisel3.experimental.BundleLiterals._
+
+import testutil._
+
+class AluMiterWithChecker(factory0 : () => AluIfc, factory1 : () => AluIfc) extends AluMiter(factory0, factory1) {
+  assert(same || (mode =/= 1.U && mode =/= 2.U && mode =/= 4.U))
+}
```

Comment:
@ekiwi Is the a good way to specify restrictions on inputs?

Reply:
Yes. You can use an `assume( ... some boolean expression ...)`

Reply:
So I guess something like:
```
assume(mode === 1.U || mode === 2.U ||  mode === 4.U)
assert(same)
```